stream/src/main/scala/org/apache/pekko/stream/javadsl/Source.scala

```diff
@@ -275,6 +275,25 @@ object Source {
   def unfoldAsync[S, E](s: S, f: function.Function[S, CompletionStage[Optional[Pair[S, E]]]]): Source[E, NotUsed] =
     new Source(scaladsl.Source.fromGraph(new UnfoldAsyncJava[S, E](s, f)))
 
+  /**
+   * Creates a sequential `Source` by iterating with the given predicate and function,
+   * starting with the given `seed` value. If the predicate returns `false` for the seed,
+   * the `Source` completes with empty.
+   *
+   * @see [[unfold]]
+   */
+  def iterate[T](seed: T, p: function.Predicate[T], f: function.Function[T, T]): Source[T, NotUsed] =
+    new Source(scaladsl.Source.iterate(seed)(elem => p.test(elem), elem => f(elem)))
+
+  /**
+   * Creates an infinite sequential `Source` by iterating with the given function,
+   * starting with the given `seed` value.
+   *
+   * @see [[unfold]]
+   */
+  def iterate[T](seed: T, f: function.Function[T, T]): Source[T, NotUsed] =
+    new Source(scaladsl.Source.iterate(seed)(ConstantFun.anyToTrue, elem => f(elem)))
```

Comment:
Should we mention in the document that Java has an API that does not require a predicate?

Reply:
I think that's is not needed, we can only add that method when we only compiles with Scala 3, the overloading issue will not be fixed in Scala 2.12/2.13 in anyway.